core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -76,7 +76,7 @@
 
     @Override
     public <ContainerType> ContainerType collectInto(Class<ContainerType> containerType) {
-        return collect(getCollectorFactoryRegistry().createCollectorFor(containerType));
+        return collect(getCollectorFactoryRegistry().createCollectorFor(getHandle().getTypeResolver().resolve(containerType)));
```

Comment:
That's very appealing.  But it removes the possibility of separating the declarative "I want this type" from the CollectorFactory's "how do I make this type" knowledge.  It also does not play well with SqlObjects, where you can have Classes annotated but not instances of a Collector.

The fact that Java's type system hurts us so badly here sucks.